hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/configuration/FlinkOptions.java

```diff
@@ -417,6 +417,19 @@ private FlinkOptions() {
       .withDescription("Enables data-skipping allowing queries to leverage indexes to reduce the search space by "
           + "skipping over files");
 
+  @AdvancedConfig
+  public static final ConfigOption<Integer> RECORD_INDEX_KEYS_MAX_COUNT = ConfigOptions
+      .key("read.record.index.keys.max.count")
+      .intType()
+      .defaultValue(8)
+      .withDescription("Record Level index statistics will be read from metadata table (MDT) for data skipping optimization,\n"
+          + "and currently the index statistics are collected by a single process, i.e., flink client for batch query or\n"
+          + "split monitor operator for streaming query. This config is used to constrain the maximum number of hoodie\n"
+          + "keys that can be read from MDT without sacrificing any performance. If the number of hoodie keys from query\n"
+          + "predicate is bigger than the maximum value, the query will fallback to not using record level index.\n"
```

Comment:
is bigger than -> is greater than

to not using record level index -> to skip the record level index filtering